### PR TITLE
Better translation code

### PR DESCRIPTION
**Pull request by NoaBablikiCertora**
translation from form format to conf file format was very ugly
This is a nicer looking solution

### CODE REVIEW

File: src/extension/utils/createConfFile.ts [hunk 1]
```diff
@@ -263,197 +102,203 @@ function processCompilerPath(file: string, pathToFile?: string): string {
...
+    )
+  }
 
   if (newForm.specObj.rules) {
     const rulesArr = newForm.specObj.rules.trim().split(',')
-    form.additionalSettings.push({
-      flag: 'rule',
-      value: rulesArr,
-    })
+    if (rulesArr.length > 0) {
```
Comment (Biced) [inline src/extension/utils/createConfFile.ts:265]: rulesArr.length

File: src/extension/utils/confFileToFormData.ts [hunk 2]
```diff
@@ -0,0 +1,372 @@
...
+    }
+    if (confFile.rule_sanity === 'advanced') {
+      specObj.ruleSanity = true
+      specObj.advancedSanity = true
+    }
+    specObj.sendOnly = true
+  }
+
+  const additionalSettings = getAdditionalSettings(confFile)
+  if (Object.keys(additionalSettings)?.length > 0) {
```
Comment (Biced) [inline src/extension/utils/confFileToFormData.ts:225]: Object.keys(additionalSettings)?.length

File: src/extension/utils/confFileToFormData.ts [hunk 3]
```diff
@@ -0,0 +1,372 @@
...
+      name: key as string,
+      value: additionalSettings[key] ? additionalSettings[key].toString() : '',
+    }))
+  }
+}
+
+export function confFileToFormData(confFile: ConfFile): NewForm {
+  const form = newForm as NewForm
+
+  if (Array.isArray(confFile.files) && confFile.files.length > 0) {
```
Comment (Biced) [inline src/extension/utils/confFileToFormData.ts:236]: confFile.files.length

File: src/extension/utils/confFileToFormData.ts [hunk 4]
```diff
@@ -0,0 +1,372 @@
...
+
+  processSolidityAttributes(confFile, form.solidityObj)
+  processSpecAttributes(confFile, form.specObj)
+
+  if (confFile.msg) {
+    form.verificationMessage = confFile.msg
+  }
+
+  // additional contracts
+  if (confFile.files && confFile.files.length > 1) {
```
Comment (Biced) [inline src/extension/utils/confFileToFormData.ts:295]: confFile.files?.length

File: src/extension/utils/confFileToFormData.ts [hunk 5]
```diff
@@ -0,0 +1,372 @@
...
+    processAdditionalContracts(confFile, form)
+  }
+  return form
+}
+
+function getContractNameFromFile(fileStr: string): string {
+  const contract = fileStr.includes(':')
+    ? fileStr.split(':').reverse()[0]
```
Comment (Biced) [inline src/extension/utils/confFileToFormData.ts:303]: please avoid using reverse() to get the last member of an array

File: src/extension/utils/confFileToFormData.ts [hunk 6]
```diff
@@ -0,0 +1,372 @@
...
+      }
+      // maybe main contract isn't explicitly mentioned
+      if (!tempForm.mainContract) {
+        const splittedPathToFile = tempForm.mainFile.split('/')
+        tempForm.mainContract = splittedPathToFile[
+          splittedPathToFile.length - 1
+        ].replace('.sol', '')
+      }
+      // link
+      if (confFile.link && confFile.link.length > 0) {
```
Comment (Biced) [inline src/extension/utils/confFileToFormData.ts:346]: confFile.link?.length

File: src/extension/utils/confFileToFormData.ts [hunk 7]
```diff
@@ -0,0 +1,372 @@
...
+        })
+      } else if (confFile.solc) {
+        if (confFile.solc.includes('/')) {
+          const solcArr = confFile.solc.split('/')
+          tempForm.compiler.ver = solcArr.pop() || ''
+          tempForm.compiler.exe = solcArr.reverse().join('/')
+        } else {
+          tempForm.compiler.ver = confFile.solc
+        }
+      }
```
Comment (Biced) [inline src/extension/utils/confFileToFormData.ts:367]: else if (confFile.solc?.includes('/')) {
          const solcArr = confFile.solc.split('/')
          tempForm.compiler.ver = solcArr.pop() || ''
          tempForm.compiler.exe = solcArr.reverse().join('/')
        } else {
          tempForm.compiler.ver = confFile.solc
      }

      tempForm.compiler.exe = solcArr.reverse().join('/') - this is why reverse() is so bad to get the last member in a string/array

**Review by Biced — CHANGES_REQUESTED** [review]
a lot of this JSON.stringify(config, null, 2) any reason you need the extra indentation? 
a lot of string/array.reverse()[0] - I don't like this approach in general.
in this PR it really shows that it is bad practice since you have to reverse it back again.

File: src/settings/utils/confFileToFormData.ts [hunk 8]
```diff
@@ -311,7 +289,7 @@ export function confFileToFormData(confFile: ConfFile): NewForm {
   }
 
   // additional contracts
-  if (confFile.files?.length > 1) {
+  if (confFile.files && confFile.files.length > 1) {
```
[the remark this answers is not in the document]
Comment (NoaBablikiCertora) [reply src/settings/utils/confFileToFormData.ts:292]: >1 not 0

File: src/extension/utils/confFileToFormData.ts (hunk 4 above)
Comment (Biced) [reply src/extension/utils/confFileToFormData.ts:295]: confFile.files?.length > 1
didn't notice the 1

**Review by Biced — CHANGES_REQUESTED** [review]
Corgi keshet, just remove the comment

**Review by Biced — CHANGES_REQUESTED** [review]

**Review by Biced — APPROVED** [review]
Corgi keshet

**Pull request merged**